flink-tests/src/test/java/org/apache/flink/test/streaming/runtime/SourceNAryInputChainingITCase.java

```diff
@@ -142,6 +153,26 @@ public void testMixedInputsWithMultipleUnionsExecution() throws Exception {
         verifySequence(result, 1L, 60L);
     }
 
+    // This tests FLINK-24300. The timeout is put here on purpose. Without the fix
+    // the tests take very long, but still finishes, so it would not hit the global timeout
+    @Test(timeout = 3_000)
```

Comment:
This should be a benchmark. I'm 99% sure that this test will be failing on azure from time to time.

Reply:
I added it as a benchmark here: https://github.com/apache/flink-benchmarks/pull/31

Reply:
Then perhaps in this PR we could only a simple UT to check multiple call to `SourceOperator#getAvailableFuture()` returns the same object?